src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -155,25 +164,25 @@ protected AbstractAddParameterCheckCodeRefactoringProvider()
                 }
 
                 // For all other parameters, add null check - updates document
-                document = await AddNullCheckAsync(document, parameter, functionDeclaration,
+                document = await AddNullCheckAsync(document, parameterSyntax, parameter, functionDeclaration,
                     (IMethodSymbol)parameter.ContainingSymbol, blockStatementOpt, cancellationToken).ConfigureAwait(false);
             }
 
             return document;
         }
 
-        private static IParameterSymbol? GetParameterAtOrdinal(int index, IReadOnlyList<SyntaxNode> parameterNodes, SemanticModel semanticModel, CancellationToken cancellationToken)
+        private static (TParameterSyntax?, IParameterSymbol?) GetParameterAtOrdinal(int index, IReadOnlyList<SyntaxNode> parameterNodes, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
             foreach (var parameterNode in parameterNodes)
             {
                 var parameter = (IParameterSymbol)semanticModel.GetRequiredDeclaredSymbol(parameterNode, cancellationToken);
                 if (index == parameter.Ordinal)
                 {
-                    return parameter;
+                    return ((TParameterSyntax)parameterNode, parameter);
```

Comment:
this worries me.  were we blindly casting before?

Reply:
It appears `parameterNodes` contains a `SeparatedSyntaxList<ParameterNode>` which was converted to an `IReadOnlyList<SyntaxNode>` within [`SyntaxGenerator.GetParameters`](https://github.com/dotnet/roslyn/blob/6aa373259bd1f19096ccf38cc8028293831a64de/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs#L2001).

I would be open to any refactoring/additional helpers which would make this code safer/more maintainable.

Reply:
I changed to simply cast the list of parameter nodes which I think will make it more clear what exactly our assumption is.